src/bin/pg_autoctl/cli_config.c

```diff
@@ -650,23 +801,12 @@ cli_keeper_config_set(int argc, char **argv)
 		/* we print out the value that we parsed, as a double-check */
 		char value[BUFSIZE];
 
-		if (!keeper_config_set_setting(&config,
-									   argv[0],
-									   argv[1]))
+		if (!cli_keeper_config_validate_and_commit(&config, argv[0], argv[1]))
 		{
 			/* we already logged about it */
 			exit(EXIT_CODE_BAD_CONFIG);
 		}
 
-		/* first write the new configuration settings to file */
-		if (!keeper_config_write_file(&config))
-		{
-			log_fatal("Failed to write pg_autoctl configuration file \"%s\", "
-					  "see above for details",
-					  config.pathnames.config);
-			exit(EXIT_CODE_BAD_CONFIG);
-		}
-
```

Comment:
Again, that change seems too invasive. The special processing of the `replication.password` setting should happen entirely elsewhere, and this code left unchanged for the other settings.